test/Mono.Linker.Tests.Cases/Inheritance.Interfaces/DefaultInterfaceMethods/Dependencies/StaticInterfaceImple.il

```diff
@@ -0,0 +1,153 @@
+
+}
+
+.class public auto ansi MyClass
+	extends [mscorlib]System.Object
+	implements IMyInterface
+{
+  .method public hidebysig specialname rtspecialname 
+          instance void  .ctor() cil managed
+  {
+    ret
+  } // end of method MyClass::.ctor
+
+  .method public hidebysig newslot virtual final 
+          instance void  Frob() cil managed
+	{
+        call void TestEntrypoint::Test()
+		ret
+	}
+}
+
+.class interface public abstract auto ansi InterfaceScenario1
+{
+  .method public newslot virtual abstract static int32 Method() cil managed noinlining
+  {
+  } // end of method Method
+} // end of class InterfaceScenario1
+.class public auto ansi BaseScenario1
+       extends [mscorlib]System.Object
+       implements InterfaceScenario1
+{
+  .method public static int32 Method() cil managed noinlining
+  {
+    .override method int32 InterfaceScenario1::Method()
+    .locals init (int32 V_O)
+    ldloca.s 0
+    initobj int32
+    ldloc.0
+    ret
+  } // end of method Method
+} // end of class BaseScenario1
+.class public auto ansi DerivedScenario1
+       extends BaseScenario1
+{
+} // end of class DerivedScenario1
+.class interface public abstract auto ansi InterfaceScenario2
+{
+  .method public newslot virtual abstract static int32 Method() cil managed noinlining
+  {
+  } // end of method Method
+} // end of class InterfaceScenario2
+.class public auto ansi BaseScenario2
+       extends [System.Runtime]System.Object
+       implements InterfaceScenario2
+{
+  .method public static int32 Method() cil managed noinlining
+  {
+    .override method int32 InterfaceScenario2::Method()
+    .locals init (int32 V_O)
+    ldloca.s 0
+    initobj int32
+    ldloc.0
+    ret
+  } // end of method Method
+} // end of class BaseScenario2
+.class public auto ansi DerivedScenario2
+       extends BaseScenario2
+       implements InterfaceScenario2
+{
+} // end of class DerivedScenario2
+.class interface public abstract auto ansi InterfaceScenario3
+{
+  .method public newslot virtual abstract static int32 Method() cil managed noinlining
+  {
+  } // end of method Method
+} // end of class InterfaceScenario3
+.class public abstract auto ansi BaseScenario3
+       extends [System.Runtime]System.Object
+       implements InterfaceScenario3
+{
+} // end of class BaseScenario3
+.class public auto ansi DerivedScenario3
+       extends BaseScenario3
+       implements InterfaceScenario3
+{
+  .method public static int32 MethodImplOnDerived() cil managed noinlining
+  {
+    .override method int32 InterfaceScenario3::Method()
+    .locals init (int32 V_O)
+    ldloca.s 0
+    initobj int32
+    ldloc.0
+    ret
+  } // end of method MethodImplOnDerived
+} // end of class DerivedScenario3
+
+.class public auto ansi TestEntrypoint
+       extends [mscorlib]System.Object
+{
+  .method public static void Test_Scenario1() cil managed noinlining
+  {
+    constrained. DerivedScenario1
+    call int32 InterfaceScenario1::Method()
```

Comment:
Just want to double check that the linker works as I think it does:
* This should be the only reference to `DerivedScenario1` the linker sees 
  * Assumes that we correctly process `constrained. ` instructions - can you double check what happens for these?
  * This ends up marking `DerivedScenario1` without anything in it
* The second instruction is treated as normal call which will mark both `InterfaceScenario1` and also `Method` on it
* Since `Method` is evaluated as a base method in a virtual hierarchy (sort of), we end up marking it on `DerivedScenario1` as well

Right?

Reply:
Yes!

- The `.constrained` instruction ends up marking `DerivedScenario1 `as you indicated above (as well as explore base types).
- The next instruction, `call`, results in marking the `InterfaceScenario1 `and `Method` in it as you expected.
- The actual method is `overriden `in BaseScenario1. Since we marked this method earlier in `MapOverrides` at `TypeMapInfo`, I assume everything works correctly since the full type hierarchy (`InterfaceScenario1`, `BaseScenario1`, `DerivedScenario1` and related methods) have been correctly marked. 
 
The post trim code looks correct.